pkg/common/cns-lib/volume/manager.go

```diff
@@ -1014,3 +1016,62 @@ func (m *defaultManager) RetrieveVStorageObject(ctx context.Context, volumeID st
 	log.Debugf("vStorageObject for volumeID: %q is %+v", volumeID, vStorageObject)
 	return vStorageObject, nil
 }
+
+// QueryVolumeAsync returns volumes matching the given filter using CnsQueryAsync API.
+func (m *defaultManager) QueryVolumeAsync(ctx context.Context, queryFilter cnstypes.CnsQueryFilter, querySelection cnstypes.CnsQuerySelection) (*cnstypes.CnsQueryResult, error) {
```

Comment:
Can we return an error if we observe vCenter client is less than the supported one from this function?
With this even when the caller forgets to make the check, we will get to know early why the call is failing.

Reply:
Makes sense. Let me add this and update the thread

Reply:
Added